### PR TITLE
feat: add ckb2021 syscalls: vm_version, current_cycles, exec

### PR DESCRIPTION
**Pull request by TheWaWaR**

### CODE REVIEW

File: src/high_level.rs [hunk 1]
```diff
@@ -554,3 +554,66 @@ pub fn find_cell_by_data_hash(data_hash: &[u8], source: Source) -> Result<Option
...
+) -> Result<u64, SysError> {
+    let index = look_for_dep_with_hash2(code_hash, hash_type)?;
+    let bounds: usize = (offset as usize) << 32 | (length as usize);
+    Ok(syscalls::exec(
+        index,
+        Source::CellDep,
+        0,
+        bounds,
+        argc,
+        argv,
```
Comment (XuJiandong) [inline src/high_level.rs]: The last element in argv should be 0. For example, when argc is 2, then argv[0], argv[1] are for arguments, argv[2] should be zero.

**Comment by XuJiandong** [comment]
When writing contracts in Rust, how can we receive the `argv` and `argc` in contracts?

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: src/high_level.rs (hunk 1 above)
Comment (TheWaWaR) [reply src/high_level.rs]: `argv` is a `CStr` array, when argc is 2, the content of `argv` may like:
- `argv[0]` = "abc\0"
- `argv[1]` = "xyzijk\n"

And there is no argv[2].

Comment (TheWaWaR) [reply src/high_level.rs]: Yes you are right, the argv[argc] is null pointer.

https://www.gnu.org/software/libc/manual/html_node/Program-Arguments.html

File: test/contract/src/exec_callee.rs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+use ckb_std::{cstr_core::CStr, debug, default_alloc, entry};
+
+use core::slice::from_raw_parts;
+
+#[no_mangle]
+pub fn main(argc: u64, argv: *const *const u8) -> i8 {
+    debug!("argc: {}", argc);
+    let args = unsafe { from_raw_parts(argv, argc as usize) };
+    for &arg in args {
```
Comment (XuJiandong) [inline test/contract/src/exec_callee.rs]: Can we have some checking on content of argv?

**Comment by TheWaWaR** [comment]
> When writing contracts in Rust, how can we receive the `argv` and `argc` in contracts?

Example contract added, please see: `test/contract/src/exec_callee.rs`

**Review by XuJiandong — APPROVED** [review]

**Pull request merged**